Pegasus/include/Geometry.hpp

```diff
@@ -9,237 +9,251 @@
 #define GLM_ENABLE_EXPERIMENTAL
 #include <glm/gtx/norm.hpp>
 #include <glm/gtx/optimum_pow.hpp>
+#include <glm/gtc/type_ptr.hpp>
 #include <glm/glm.hpp>
 
 #include <algorithm>
-#include <array>
-#include <functional>
+#include <utility>
 #include <limits>
 #include <memory>
-#include <cmath>
-#include <unordered_map>
+#include <array>
 #include <vector>
-#include <utility>
-#include <cstdint>
+#include <unordered_map>
+
+#include "Pegasus/include/Math.hpp"
 
 namespace pegasus
 {
 namespace geometry
 {
+/**
+ * @brief Base shape class
+ */
 class Shape
 {
 public:
-    explicit Shape(glm::dvec3 const& centerOfMass);
-
-    void SetCenterOfMass(glm::dvec3 const& centerOfMass);
+    glm::dvec3 centerOfMass;
 
-    glm::dvec3 const& GetCenterOfMass() const;
+    Shape() = default;
 
-private:
-    glm::dvec3 m_centerOfMass;
+    explicit Shape(glm::dvec3 const& centerOfMass);
 };
 
-enum class SimpleShapeType : uint32_t
+/**
+ * @brief Generic class for representing shapes that could be described as a simple parametric or quadric surface
+ */
+class SimpleShape : public Shape
 {
-    PLANE,
-    TRIANGLE,
-    SPHERE,
-    CONE,
-    CYLINDER,
-    CAPSULE,
-    BOX,
-    NONE
+public:
+    enum class Type : uint8_t
+    {
+        RAY,
+        PLANE,
+        TRIANGLE,
+        SPHERE,
+        CONE,
+        CYLINDER,
+        CAPSULE,
+        BOX,
+        NONE
```

Comment:
There is a `Type type` field in `SimpleShape` class, which will initialize anyway without any meaning. I'd rather it be `NONE` than 0 or any arbitrary shape type.